src/commands/issue.ts

```diff
@@ -0,0 +1,25 @@
+import { IMigrationContext } from '../migration-context';
+import forEachRepo from '../util/for-each-repo';
+
+export default async (context: IMigrationContext) => {
```

Comment:
If we open an issue and want to update it later, how would we go about doing that?  A basic use case would be a typo but can be more substantial, for example additional references or detail that was not initially captured when the issue was first published.

Similarly, I'm thinking labels should be updateable.

Reply:
Yes, I will add another feature on top of this to get the issue with issue # and update it.

Reply:
I’m curious to learn more about the implementation details for this.  How are we thinking of persisting state so that we can update issues and labels?

Reply:
the issue can be updated with issue # and it is challenging to track the issue number for multiple repositories. I will think of different way to persist the state

Reply:
Take a look at Shepherd's existing persistence before rolling your own (unless, of course, there's a need).

Reply:
Persisted-data tracks checkout repos under repos.json.  We can use a similar strategy to track issues, perhaps under an issues.json.  When an issue is created, using the GitHub Create Issues API, on success, we get a response that contains the Issue number.  We can track this number in a map of repos.  For example:

```
[
repo1: {
   issueNumber: 1
},
repo2: {
   issueNumber: 3
},
repo3: {
   issueNumber: 1
}
]
```

Migrations may not necessarily be 1:1 with issue creation.  That is, a migration can create more than one issue.  So the above won't suffice.  To accomodate multiple issues, we can store the data as follows:

```
[
repo1: {
  issues: [
     {
        issueNumber: 1, 
        issueKey: identifier1 // from migration spec; each issue will have a unique identifier
      }
   ]
},
repo2: {
   issues: [
     {
         issueNumber: 3, 
         issueKey: identifier2
     }
  ]
},
repo3: {
   issues: [
      {
         issueNumber: 1, 
         issueKey: identifier1
      },
      {
         issueNumber: 3, 
         issueKey: identifier2
      }
    ]
}
]
```

Persisting the data in this way will enable us to update issues as needed.

Reply:
This is just a suggestion and I haven't considered alternative data structures and so this might not be optimal.  Note, I'm also sure we can add to the repos.json file but I haven't assessed whether that's going to benefit us in the long term or not.  Please evaluate accordingly.

Reply:
yes i also had a same thought however, the create and update issue should be a seperate flow and we can match the title and fetch the issue number to update the same issue

Reply:
Why wouldn’t we be able to update the title?

Reply:
Actually,  if the issue identifier with issue number can be persisted while creating an issues across multiple repos, the issue identifier can be repository name along with issue #- then if i post another set of issues to same set of repos, the issues.json will be overwritten, with new issue numbers. In that case, i can update the title, message, and labels. 

Solution#1: Save the state of posted issue with title + repo + issue# 

In this case, the title can't be updated, and more than 1 issue can be posted/updated on a repo, and each issue can be persisted in an issues.json

Solution#2: 

Persist the state of posted issues with repo name + issue #

In this case, the title can be updated, but only 1 issue can be persisted for a repo.

Reply:
Ah, a likely thing I could imagine wanting (that is relevant to persisting), is adding an additional comment to the issue that was created. 

For example, the initial issue might be about some API they need to change. I might want to follow that up with a "There is 30 more days until this API is removed" or something like that.

Reply:
We can have separate workflows for adding comments on the open issue. The lifecycle for posting comments on issue workflow should continue the "create issue" workflow. 

Create issue -> save the state of posted issues -> Read/update/delete issues & post comment on issue.

Reply:
If we choose to support comments, we would certainly need a reference to the issues on which to comment.  We would likely need to maintain state and provide a view of open issues, their state, and repo.  

This use case highlights a more general feature gap wherein developers should have view access to Shepherd's state.  This would help them better understand what actions to take through Shepherd (e.g., list PRs associated with a migration and their state).

cc @TheSavior